### DISCUSSION TITLE
how to do a security audit on a software brick, like symfony for example

### DISCUSSION THREAD

**Discussion by aragon5956** [symfony/symfony#59119] [Q&A]
### Description

hello,
 how to do a security audit on a software brick, like those of symfony? is what you need to create a docker container and/or create a web project to test the software brick , in question ?

bonjour,
 comment faire un audit de sécurité sur une brique logicielle, comme celles de symfony ?est ce qu'il faut créer un  container docker et/ou créer un projet web pour tester la brique logicielle , en question ?

**Comment by nicolas-grekas** [comment]
I'm sorry but your question is too vague. Doing a security audit means identifying security scenarios, threat models, etc. Nothing specific about Symfony. Of course, once you have answered those questions, you can write test cases, functional ones, etc.

**Reply by aragon5956** [reply]
non c'est pas trop vague, si je fais un `composer audit` , je trouve plein de failles non corrigées , et j'aimerais faire la même chose pour les briques doctrine

**Comment by nicolas-grekas** [comment]
Now that you mentioned composer audit, this might be clearer.
There's nothing to do actually: composer audit will report issues with any packages, not only Symfony ones (provided maintainers report security issues properly, which most do.)

**Reply by aragon5956** [reply]
???justement il ya des paquets symfony à corriger !!There are packages symfony to be corrected!

**Reply by nicolas-grekas** [reply]
`compose update` to the rescue.
Please use English so that others that have the same need could use this discussion later on.

**Comment by aragon5956** [comment]
![auditpaquetsDeux](https://github.com/user-attachments/assets/f9ece029-4e51-4999-931d-ee06f011987a)
![auditPaquetsUn](https://github.com/user-attachments/assets/938b9b39-d909-4cec-8a07-f26d8834be39)

**Reply by nicolas-grekas** [reply]
Did you run composer update?
What's the output of composer show?

**Reply by aragon5956** [reply]
laissez tomber
